NAMESPACE

```diff
@@ -0,0 +1,103 @@
+# Generated by roxygen2: do not edit by hand
+
+export(ashm_multi_rainbow_plot)
+export(ashm_rainbow_plot)
+export(comp_report)
+export(copy_number_vaf_plot)
+export(fancy_alignment_plot)
+export(fancy_circos_plot)
+export(fancy_cnbar)
+export(fancy_ideogram)
+export(fancy_multisamp_ideogram)
+export(fancy_propcov_plot)
+export(fancy_proportions_plot)
+export(fancy_qc_plot)
+export(fancy_snv_chrdistplot)
+export(fancy_sv_sizedens)
+export(fancy_v_chrcount)
+export(fancy_v_count)
+export(fancy_v_sizedis)
+export(focal_cn_plot)
+export(get_gambl_colours)
+export(get_mutation_frequency_bin_matrix)
+export(plot_sample_circos)
+export(prettyChromoplot)
+export(prettyCoOncoplot)
+export(prettyForestPlot)
+export(prettyGeneCloud)
+export(prettyOncoplot)
+export(prettyRainfallPlot)
+export(pretty_lollipop_plot)
+export(splendidHeatmap)
+export(theme_Morons)
+import(ComplexHeatmap)
+import(RCircos)
+import(RColorBrewer)
+import(broom)
+import(circlize)
+import(cowplot)
+import(data.table, except = c("last", "first", "between", "transpose"))
+import(dplyr)
+import(forcats)
+import(g3viz)
+import(ggbeeswarm)
+import(ggplot2)
+import(ggpubr, except = "get_legend")
+import(ggrepel)
+import(ggsci)
+import(gridExtra, except = "combine")
+import(maftools)
+import(plotly, except = c("last_plot", "add_heatmap", "export"))
+import(purrr)
+import(readr)
+import(stringr)
+import(tibble)
+import(tidyr)
+import(wordcloud)
+importFrom(GAMBLR,annotate_sv)
+importFrom(GAMBLR,assign_cn_to_ssm)
+importFrom(GAMBLR,calc_mutation_frequency_sliding_windows)
+importFrom(GAMBLR,cnvKompare)
+importFrom(GAMBLR,collate_results)
+importFrom(GAMBLR,fread_maf)
+importFrom(GAMBLR,gene_to_region)
+importFrom(GAMBLR,get_cn_segments)
+importFrom(GAMBLR,get_combined_sv)
+importFrom(GAMBLR,get_gambl_metadata)
+importFrom(GAMBLR,get_manta_sv)
+importFrom(GAMBLR,get_sample_cn_segments)
+importFrom(GAMBLR,get_ssm_by_region)
+importFrom(GAMBLR,get_ssm_by_sample)
+importFrom(GAMBLR.helpers,check_config_value)
+importFrom(GAMBLR.helpers,coding_vc)
+importFrom(GAMBLR.helpers,collate_qc_results)
+importFrom(GAMBLR.helpers,map_metadata_to_colours)
+importFrom(GAMBLR.helpers,region_to_chunks)
+importFrom(GAMBLR.helpers,standardize_chr_prefix)
+importFrom(GAMBLR.helpers,subset_cnstates)
+importFrom(ggthemes,theme_foundation)
+importFrom(grDevices,col2rgb)
```

Comment:
is `grDevices` a default package, i.e a package that gets loaded automatically R? If so, is it still necessary to state what functions from that package we want? The same question applies to other default(?) R packages such as `utils`, `stats`, `methods`, `grid`, `graphics`.

Reply:
I'm not sure. I added them just for `devtools::check()` to be happy.

Reply:
Do you have by any chance the error message it was showing? It seems to only be used in the get_gambl_colours and is not prepended with grDevices:: so should be ok to go. Also, doesn't seem to show an error in the current master of GAMBLR but maybe something changes when we use the per-function file instead of one big script

Reply:
Actually, `devtools::check()` doesn't say it is an error, but just a note. But it's not the case that I'm trying to fix notes. It's just the way how my function to fix dependencies works.
This function takes a `devtools::check()` output and looks for lines like
```
ashm_multi_rainbow_plot: no visible global function definition for
  ‘get_gambl_metadata’
```
following the example, the function uses a given list of package candidates (`GAMBLR`, `GAMBLR.helpers`, `GAMBLR.data`, etc) to find from which package `get_gambl_metadata` is, and then runs `usethis::use_import_from("GAMBLR", "get_gambl_metadata")`.
Since `ashm_multi_rainbow_plot` was moved from `GAMBLR` to `GAMBLR.viz`, of course it doesn't have a `#' @import GAMBLR` or a `#' @importFrom GAMBLR get_gambl_metadata` tag in it's documentation. So, the easiest solution that I found was to use `usethis::use_import_from` to add `importFrom(GAMBLR,get_gambl_metadata)` to the NAMESPACE of GAMBLR.viz and, most importantly, to create the file `GAMBLR.viz-package.R`, where all `#' @importFrom packageName functionName` are saved.

However, my function also takes those lines in `devtools::check()` output that mention a function from an R-base package. If it's important to not import any of these functions, I can change my function to check if the function is from an R-base package (`rownames(installed.packages(priority="base"))`) and, if so, don't import it.

Reply:
Thanks for explaining! I think it is fine to keep it like this here. For the other packages, and other scripts that would be in thematic packages, I don't think we need to focus on notes, either this one or any other note that comes up. They can be left out and we can always make them pretty when we have time or need to focus on it. For now, let's keep it as is and move on to making this package live :computer: